src/ui/ResultLink/ResultLink.ts

```diff
@@ -478,7 +479,7 @@ export class ResultLink extends Component {
 
   private getResultUri(): string {
     if (this.options.hrefTemplate) {
-      return this.parseStringTemplate(this.options.hrefTemplate);
+      return StringUtils.buildStringTemplateFromResult(this.options.hrefTemplate, this.result);
```

Comment:
If a user from a given client is capable of choosing the template, this is a potential site for an XSS injection.

Reply:
@olamothe thoughts on that? This function is used in ResultLink also.

Reply:
If you can set that option, you can modify anything in the page anyway, so yes it's possible to have XSS here, like anything else.